src/main/java/org/spongepowered/common/data/processor/common/BreakablePlaceableUtils.java

```diff
@@ -0,0 +1,99 @@
+/*
+ * This file is part of Sponge, licensed under the MIT License (MIT).
+ *
+ * Copyright (c) SpongePowered <https://www.spongepowered.org>
+ * Copyright (c) contributors
+ *
+ * Permission is hereby granted, free of charge, to any person obtaining a copy
+ * of this software and associated documentation files (the "Software"), to deal
+ * in the Software without restriction, including without limitation the rights
+ * to use, copy, modify, merge, publish, distribute, sublicense, and/or sell
+ * copies of the Software, and to permit persons to whom the Software is
+ * furnished to do so, subject to the following conditions:
+ *
+ * The above copyright notice and this permission notice shall be included in
+ * all copies or substantial portions of the Software.
+ *
+ * THE SOFTWARE IS PROVIDED "AS IS", WITHOUT WARRANTY OF ANY KIND, EXPRESS OR
+ * IMPLIED, INCLUDING BUT NOT LIMITED TO THE WARRANTIES OF MERCHANTABILITY,
+ * FITNESS FOR A PARTICULAR PURPOSE AND NONINFRINGEMENT. IN NO EVENT SHALL THE
+ * AUTHORS OR COPYRIGHT HOLDERS BE LIABLE FOR ANY CLAIM, DAMAGES OR OTHER
+ * LIABILITY, WHETHER IN AN ACTION OF CONTRACT, TORT OR OTHERWISE, ARISING FROM,
+ * OUT OF OR IN CONNECTION WITH THE SOFTWARE OR THE USE OR OTHER DEALINGS IN
+ * THE SOFTWARE.
+ */
+package org.spongepowered.common.data.processor.common;
+
+import com.google.common.collect.Sets;
+import net.minecraft.item.ItemStack;
+import net.minecraft.nbt.NBTTagCompound;
+import net.minecraft.nbt.NBTTagList;
+import net.minecraft.nbt.NBTTagString;
+import org.spongepowered.api.block.BlockType;
+import org.spongepowered.api.data.value.ValueContainer;
+import org.spongepowered.common.Sponge;
+import org.spongepowered.common.data.util.NbtDataUtil;
+
+import java.util.Optional;
+import java.util.Set;
+
+public final class BreakablePlaceableUtils {
+
+    public static boolean set(ValueContainer<?> holder, String nbtKey, Set<BlockType> value) {
+    	if (!(holder instanceof ItemStack)) {
+    		return false;
+    	}
+    	ItemStack stack = (ItemStack) holder;
```

Comment:
I would got to the effect of declaring local variables `final` as much as possible.